Fundamentals/Backoffice/Property-Editors/Built-in-Property-Editors/Block-Editor/build-custom-view-for-blocks.md

```diff
@@ -0,0 +1,133 @@
+
+Create two files within the CustomBlockView file: `package.manifest` and `customBlock.controller.js`.
+ 
+Add the following JSON to the `package.manifest` file:
+
+```json
+{
+  "javascript": [
+    "~/App_Plugins/CustomBlockView/customBlock.controller.js"    
+  ]  
+}
+```
+
+:::note
+Umbraco will parse all `package.manifest` files and load any resources they reference into the backoffice during startup.
+:::
+
+The second file, `customBlock.controller.js`, will be used to register the 'customBlockController' defined using the `ng-controller` attribute in your custom view.
+
+To register the controller, add the following lines of code:
+
+```javascript
+angular.module("umbraco").controller("customBlockController", function ($scope) {
+// you can do your custom functionality here!
+});
+```
+
+### Example: Displaying an image from a Media Picker
+
+Your block may enable you to 'pick' an image for use as the background for a particular block. If you try to display this image directly in the view using `block.data.image`, you will see the unique ID and not the image.
+
+You will need to use the ID in our custom AngularJS controller in order to get the `ImageUrl` to display in our backoffice Block Editor View.
+
+With the setup of files above, you need to amend the `customBlock.controller.js` file, by injecting the `mediaResource` to retrieve the image from the ID:
+
+```javascript
+angular.module("umbraco").controller("customBlockController", function ($scope, mediaResource) {
+
+    //your property is called image so the following will contain the udi:
+        var imageUdi = $scope.block.data.image;
+    //the mediaResource has a getById method:
+        mediaResource.getById(imageUdi)
+            .then(function (media) {
+                console.log(media);
+                //set a property on the 'scope' called imageUrl for the returned media object's mediaLink
+                $scope.imageUrl = media.mediaLink;
+    });    
+});
+```
+
+Update the Custom View to use the `imageUrl` property to display the image:
+
+```html
+<div ng-controller="customBlockController" ng-click="block.edit()">
+    <h2 ng-bind="block.data.headline"></h2>
+    <img src="{{imageUrl}}" />
+    <p ng-bind="block.data.description"></p>
+</div>
+```
+
+If you need to use a specific crop, you can inject the `imageUrlGeneratorResource` resource, which has a `getCropUrl(mediaPath, width, height, imageCropMode, animationProcessMode)` method:
+
+```javascript
+angular.module("umbraco").controller("customBlockController", function ($scope, mediaResource,imageUrlGeneratorResource) {
+
+    //your property is called image so the following will contain the udi:
+        var imageUdi = $scope.block.data.image;
+    //the mediaResource has a getById method:
+        mediaResource.getById(imageUdi)
+            .then(function (media) {
+                imageUrlGeneratorResource.getCropUrl(media.mediaLink, 150, 150).then(function (cropUrl) {
```

Comment:
Do we have a method in getCropUrl to give it a crop alias that is on the image, be it local to the content or directly on the media item itself @nielslyngsoe ?

Reply:
Ehm, i dont know.. would have to pass that one to someone else to check.